rllib/env/single_agent_env_runner.py

```diff
@@ -304,7 +307,18 @@ def _sample_timesteps(
                         truncated=truncateds[env_index],
                         extra_model_outputs=extra_model_output,
                     )
-
+                    # We have to perform an extra env-to-module pass here, just in case
```

Comment:
Important fix: The very last obs must also be sent through the connector, just in case the user performs some writing back into the episode (trajectory). Otherwise, for example bootstrapping computations at the episode truncations would have distorted results (or the forward pass would fail entirely due to a format change).